contract/r/gnoswap/staker/staker.gno

```diff
@@ -552,8 +552,19 @@ func CollectReward(positionId uint64, unwrapResult bool) (string, string, map[st
 func UnStakeToken(positionId uint64, unwrapResult bool) (string, string, string) { // poolPath, token0Amount, token1Amount
 	assertOnlyNotHalted()
 
-	// unStaked status
+	// onwer check
 	deposit := deposits.Get(positionId)
+	caller := getPrevAddr()
+	if deposit.owner != caller {
```

Comment:
Please add a unit test or leave a comment if not needed.

Reply:
left the comment because the checking condition is clear.

Also, this function is difficult to write unit tests for, it seems more appropriate to test it in a live environment.
https://github.com/gnoswap-labs/gnoswap/pull/626/commits/9cc8da30d45945aff3bb831fa5a71efebf7249de